src/ng/compile.js

```diff
@@ -2052,24 +2052,30 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
           addTextInterpolateDirective(directives, node.nodeValue);
           break;
         case NODE_TYPE_COMMENT: /* Comment */
-          try {
-            match = COMMENT_DIRECTIVE_REGEXP.exec(node.nodeValue);
-            if (match) {
-              nName = directiveNormalize(match[1]);
-              if (addDirective(directives, nName, 'M', maxPriority, ignoreDirective)) {
-                attrs[nName] = trim(match[2]);
-              }
-            }
-          } catch (e) {
-            // turns out that under some circumstances IE9 throws errors when one attempts to read
-            // comment's node value.
-            // Just ignore it and continue. (Can't seem to reproduce in test case.)
-          }
+          collectCommentDirectives();
           break;
       }
 
       directives.sort(byPriority);
       return directives;
+
+      function collectCommentDirectives() {
```

Comment:
Ok,

I think that comments from @dcherman and @jbedard are not completely true. Although, may be I have some reasonable doubts, but I expose the following:

- functions are statically parsed in the beginning of the execution: it does not matter if they are nested or whatever. I'm quite sure that it is being doing by this way because even the old Modula II did that,

- There is no new Function(bodyString) is executed neither eval("function() {...}"): so, I do not see where the compiler creates the new function instance for each and every one though

- Closures are a old concepts that can be optimized easily: In this case the compiler can analyse statically the code, it can see that collectCommentDirectives has no other use outside collectDirectives, it does not define any new variable, it does not has arguments, so there is no reason to not reuse the stack from collectDirectives. It looks a lot like C nested functions, so it is possible to the compiler to apply the same optimization techniques. I'm not sure if compilers are doing it, but it is a very simple optimisation, it can be applied in many different situations, so it is very likely that it might be already implemented

- this function is only called when comments are presented: in the worst case that there are no optimizations, the closure creation will only happen when comments are parsed. It turns out that in production environments in which people is concerned by performance, there should be no comments at all (they are a waste of bandwidth and parse time)

In addition, the strong point of having this implemented by a nested function is to make it easier to understand, and easier to maintain: the implementation is close to where it is called, it is the same code that was before, it uses the same variables and structure that the parent function. And basically this function, is only a protection, has no other kind of meaning.

Any way,

- I'll try to do some performance benchmarks about this case with synthetic cases (it is a pity that jsperf is offline) just to make sure that there is no a performance problem (in my tests in makes no difference to have this function in this level or in another level with clausure variables passed by value, but it is almost not called)

- If it is just me that thinks that it is easier to understand and maintain implemented by this way, I can change it, no problem